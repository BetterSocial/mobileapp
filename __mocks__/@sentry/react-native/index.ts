export const addBreadcrumb = jest.fn();
