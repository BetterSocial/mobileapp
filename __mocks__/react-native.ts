import * as RN from 'react-native';

RN.Animated.timing = () => ({
    start: () => jest.fn()
})

module.exports = RN