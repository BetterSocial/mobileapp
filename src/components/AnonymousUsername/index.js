import * as React from 'react';
import {StyleSheet, Text} from 'react-native';

import {POST_VERSION} from '../../utils/constants';
import {colors} from '../../utils/colors';
import {fonts} from '../../utils/fonts';

const styles = StyleSheet.create({
  feedUsername: {
    fontFamily: fonts.inter[600],
    fontWeight: 'bold',
    fontSize: 14,
    lineHeight: 16.94,
    color: colors.black,
    flex: 1
  }
});

/**
 * @typedef {Object} AnonymousUsernameComponentProps
 * @property {string} version
 * @property {AnonUserInfoTypes} anonUserInfo
 */

/**
 *
 * @param {AnonymousUsernameComponentProps} props
 */
const AnonymousUsername = (props) => {
  const {version, anonUserInfo} = props;

  if (version >= POST_VERSION) {
    return (
      <Text
        testID="newVersion"
        style={styles.feedUsername}>{`${anonUserInfo.colorName} ${anonUserInfo.emojiName}`}</Text>
    );
  }

  return <Text style={styles.feedUsername}>Anonymous</Text>;
};

export default AnonymousUsername;
