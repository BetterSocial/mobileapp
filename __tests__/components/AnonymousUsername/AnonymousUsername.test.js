import * as React from 'react';
import {render} from '@testing-library/react-native';

import AnonymousUsername from '../../../src/components/AnonymousUsername';

describe('AnonymousUsername test', () => {
  it('Should match version 1 snapshot', () => {
    const {toJSON, getByTestId} = render(
      <AnonymousUsername version={1} anonUserInfo={{colorName: 'red', emojiName: 'hourse'}} />
    );
    expect(toJSON).toMatchSnapshot();
    console.log(getByTestId('v1').props, 'lapuk');
    expect(getByTestId('v1').props.children).toEqual('Anonymous');
  });

  it('Should match version 1 snapshot', () => {
    const {toJSON, getByTestId} = render(
      <AnonymousUsername version={2} anonUserInfo={{colorName: 'red', emojiName: 'horse'}} />
    );
    expect(toJSON).toMatchSnapshot();
    expect(getByTestId('v2').props.children).toEqual('red horse');

  });
});
