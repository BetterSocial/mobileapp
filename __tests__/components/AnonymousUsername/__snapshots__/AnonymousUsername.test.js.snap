// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnonymousUsername test Should match version 1 snapshot 1`] = `[Function]`;

exports[`AnonymousUsername test Should match version 1 snapshot 2`] = `[Function]`;
